File: src/main/java/tk/hiddenname/smarthome/model/signal/Signal.java
package tk.hiddenname.smarthome.model.signal;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;

@AllArgsConstructor
@EqualsAndHashCode(of = {"id"})
abstract class Signal {

    @Getter
    public final Long id;
}

File: src/main/kotlin/tk/hiddenname/smarthome/model/signal/Signal.kt
package tk.hiddenname.smarthome.model.signal

import javax.persistence.MappedSuperclass
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull

@MappedSuperclass
abstract class Signal(
    @field:NotNull(message = "hardware id should be specified")
    @field:Min(value = 0, message = "Hardware Id should be equal or rather than 0")
    var hardwareId: Long? = null
)